contributors/design-proposals/taint-node-by-condition.md

```diff
@@ -0,0 +1,40 @@
+# Taints Node according to NodeConditions
+
+@k82cn, @gmarek, @jamiehannaford,  Jul 15, 2017
+
+## Relevant issues:
+
+* https://github.com/kubernetes/kubernetes/issues/42001
+* https://github.com/kubernetes/kubernetes/issues/45717
+
+## Motivation
+In kubernetes 1.8 and before, there are six Node Conditions, each with three possible values: True, False or Unknown. Kubernetes components modify and check those node conditions without any consideration to pods and their specs. For example, the scheduler will filter out all nodes whose NetworkUnavailable condition is True, meaning that pods on the host network can not be scheduled to those nodes, even though a user might want that.  The motivation of this proposal is to taint Nodes based on certain conditions, so that other components can leverage Tolerations for more advanced scheduling.
+
+## Functional Detail
+Currently (1.8 and before), the conditions of nodes are updated by the kubelet and Node Controller. The kubelet updates the value to either True or False according to the node’s status. If the kubelet did not update the value, the Node Controller will set the value to Unknown after a specific grace period. 
+
+In addition to this, with taint-based-eviction, the Node Controller already taints nodes with either NotReady and Unreachable if certain conditions are met. In this proposal, the Node Controller will use additional taints on Nodes. The new taints are described below:
+
+| ConditionType      | Condition Status   |Effect        | Key      |
+| ------------------ | ------------------ | ------------ | -------- | 
+|Ready               |True                | -            | |
+|                    |False               | NoExecute    | node.alpha.kubernetes.io/notReady           |
```

Comment:
What is the intent of putting alpha in the name? What is the plan for transitioning this to beta non disruptively? Is there a reason not to use the taint without alpha in the name and simply gate the function in the node controller with a feature gate?